Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/negotiation/models/Agreements.kt

```diff
@@ -22,11 +22,6 @@ data class Agreements(@Id @GeneratedValue val ContractID:UUID,
     /*@OneToMany(fetch = FetchType.LAZY, mappedBy = "agreements", orphanRemoval = true, cascade = [CascadeType.ALL])
     var messages: List<Messages>? = emptyList()*/
 
-    @ManyToOne(fetch = FetchType.LAZY)
-    //@JoinColumn(name="PublicWalletID")
-    lateinit var partyA: User
-
-    @ManyToOne(fetch = FetchType.LAZY)
-    //@JoinColumn(name="PublicWalletID")
-    lateinit var partyB: User
+    @ManyToMany(mappedBy = "agreements", cascade = [CascadeType.ALL])
+    val users: MutableSet<User> = mutableSetOf()
```

Comment:
Does this mean you're no longer differentiating between user A and B? I can see why you did it, but it will likely cause problems with Smart Contract

Reply:
So according to all business logic we have two parties. The database just does not have this restriction. This was the only clean solution me and Tristan could come up with

Reply:
Does it affect the API? If not, then it's fine

Reply:
Not at all. The API request/response objects have stayed unchanged

Reply:
All right, then it's fine by me